libs/ballot-interpreter/benchmarks/benchmarking.ts

```diff
@@ -0,0 +1,251 @@
+/* eslint-disable vx/gts-safe-number-parse */
+import { assertDefined, iter, range } from '@votingworks/basics';
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'fs';
+import { dirname, join } from 'path';
+
+const { UPDATE_BENCHMARKS, BENCHMARKS_ENV: BENCHMARK_ENV } = process.env;
+const ENV = BENCHMARK_ENV ?? 'development-m1-macbook-pro';
+
+async function measureTime(func: () => Promise<void>): Promise<Milliseconds> {
+  const start = process.hrtime.bigint();
+  await func();
+  const end = process.hrtime.bigint();
+
+  return Number((end - start) / BigInt(10_000)) / 100;
+}
+
+type Milliseconds = number;
+
+interface BenchmarkStats {
+  min: Milliseconds;
+  max: Milliseconds;
+  mean: Milliseconds;
+  median: Milliseconds;
+  variance: Milliseconds;
+  standardDeviation: Milliseconds;
+  marginOfError: Milliseconds;
+}
+
+interface BenchmarkResults {
+  measurements: Milliseconds[];
+  stats: BenchmarkStats;
+}
+
+function computeStats(measurements: Milliseconds[]): BenchmarkStats {
+  const min = Math.min(...measurements);
+  const max = Math.max(...measurements);
+  const mean = iter(measurements).sum() / measurements.length;
+  // See https://www.mathsisfun.com/data/standard-deviation.html
+  const variance =
+    iter(measurements)
+      .map((x) => (x - mean) ** 2)
+      .sum() /
+    (measurements.length - 1);
+  const standardDeviation = Math.sqrt(variance);
+  const median = assertDefined(
+    [...measurements].sort()[Math.floor(measurements.length / 2)]
+  );
+  // 99% confidence interval
+  // See https://www.mathsisfun.com/data/confidence-interval.html
```

Comment:
`mathsisfun.com` 😆

Reply:
Honestly had the most approachable explanations of any resource I found lol